File: cuboid_route.py
# problem 86

import itertools
import math

M = 100
epsilon = 1e-3
count = 0

for j, k, l in itertools.combinations_with_replacement(range(1,M + 1), r=3):
  path = math.sqrt((j + k)**2 + l**2)
  if path - int(path) < epsilon:
    count += 1
'''
for jk, l in itertools.product(range(2, 2*M + 1), range(1,M + 1)):
  path = math.sqrt((jk)**2 + l**2)
  if path - int(path) < epsilon:
    count += (jk - l + 1 - 1)/2 + 1
'''
print count

File: cuboid_route.cc
// Problem 86

#include <iostream>
#include <cmath>

using namespace std;

int main(void) {
  int M = 1817;
  double epsilon = 1e-9;
  int count = 0;
  for (int j = 1; j <= M; ++j) {
    for (int k = j; k <= M; ++k) {
      for (int l = k; l <= M; ++l) {
        double path = sqrt((j+k)*(j+k) + l*l);
        if (path - int(path) < epsilon) {
          count++;
        }
      }
    }
  }
  cout << count << endl;
  return 0;
}